vtysh/vtysh.c

```diff
@@ -2393,23 +2392,12 @@ DEFUNSH(VTYSH_REALLYALL, vtysh_disable, vtysh_disable_cmd, "disable",
 }
 
 DEFUNSH(VTYSH_REALLYALL, vtysh_config_terminal, vtysh_config_terminal_cmd,
-	"configure [terminal]",
-	"Configuration from vty interface\n"
-	"Configuration terminal\n")
-{
-	vty->node = CONFIG_NODE;
-	return CMD_SUCCESS;
-}
-
-DEFUNSH(VTYSH_REALLYALL, vtysh_config_terminal_file_lock,
-	vtysh_config_terminal_file_lock_cmd,
-	"configure terminal file-lock",
+	"configure [terminal [file-lock]]",
```

Comment:
Using `file-lock` is not only needed for config reading, but also the way to work with FRR in explicit-commit mode instead of implicit-commit-after-every-command mode. However, there are two problems:
- this explicit-commit mode works only for mgmtd-converted daemons
- `file-lock` naming is confusing and doesn't explain that it enables explicit-commit mode

So maybe you're right and we should hide it until all daemons are converted to mgmtd. But also there's an option to rename `file-lock` to something like `locked` or `with-commit` and leave it not hidden.